docs/content/concepts/terminology.md

```diff
@@ -1,100 +1,192 @@
-similar to a virtual cluster as defined by sig-multicluster, but is able to amortize the cost of a new cluster to be
-zero or near-zero memory and storage so that we can create tens of millions of empty clusters cheaply.
+A logical cluster is a way to subdivide a kcp server and its `etcd` datastore into multiple clusters without requiring
+multiple API server and etcd instances. A logical cluster is able to amortize the cost of a new cluster to be near-zero
+memory and storage, so that we can create a huge number of empty clusters cheaply.
+
+Each logical cluster has it's own set of APIs installed, it's own objects, and different access semantics and
+policies, i.e. RBAC rules. In other words, each logical cluster is isolated from other logical clusters.
+
+This is accomplished by adding an additional attribute to an object's identifier in etcd and kcp server to associate
+object with its logical cluster. On the storage level, the regular Kubernetes API server identifies objects using
+(group, version, resource, optional namespace, name). The kcp server enriches this identifier with logical cluster
+name, so we have (group, version, resource, **logical cluster name**, optional namespace, name).
+
+A logical cluster provides Kubernetes-cluster-like HTTPS/CRUD endpoints, i.e. endpoints that typical Kubernetes client
+tooling (e.g. client-go, controller-runtime, and others) can interact with as they would with regular Kubernetes
+clusters. Using the appropriate kubeconfig file, a user can run `kubectl get all -A` to return all objects in all
+namespaces in that concrete logical cluster.
 
-A logical cluster is a storage level concept that adds an additional attribute to an object’s identifier on a
-kube-apiserver.  Regular servers identify objects by (group, version, resource, optional namespace, name).  A logical
-cluster enriches an identifier: (group, version, resource, **logical cluster name**, optional namespace, name).
+To a user, a logical cluster appears to be a Kubernetes cluster without all the container orchestration specific
+resources (e.g. Pods, ReplicaSets, Deployments...). It has its own discovery, its own OpenAPI spec, and follows the
+Kubernetes-like constraints about uniqueness of Group-Version-Resource and its behavior. For example, it's not
+possible to have two identical GVRs with different schemas in one logical cluster, but it's possible to have two
+identical GVRs with different schemas in different logical clusters.
+
+However, in the most of cases, you'll never create a logical cluster on its own. Instead, you would use an abstraction,
+such as Workspace, to create logical cluster.
+
+!!! note
+    There could be multiple different models that result in logical clusters being created, with different policies or
+    lifecycle, but Workspace is intended to be the most generic representation of the concept with the broadest possible
+    utility to anyone building control planes.
```

Comment:
maybe a too strong statement. Not sure it is most generic. Would rather call it the canonical way.

Reply:
Is this any better?

> There could be multiple different models that result in logical clusters being created, with different policies or
    lifecycle. The Workspace is intended to be the most canonical way to create logical clusters for anyone willing
    to build control planes.